catroid/src/main/java/org/catrobat/catroid/common/BroadcastSequenceMap.java

```diff
@@ -28,43 +28,43 @@
 import java.util.HashMap;
 import java.util.Map;
 
-public final class BroadcastSequenceMap {
-	private static Map<String, HashMap<String, ArrayList<SequenceAction>>> broadcastSequenceMap = new HashMap<>();
+public class BroadcastSequenceMap {
+	protected Map<String, Map<String, ArrayList<SequenceAction>>> broadcastSequenceMap = new HashMap<>();
 
-	private BroadcastSequenceMap() {
-		throw new AssertionError();
+	public boolean containsKey(String key, String sceneName) {
+		return broadcastSequenceMap.containsKey(sceneName) && broadcastSequenceMap.get(sceneName).containsKey(key);
 	}
 
-	public static boolean containsKey(String key, String sceneName) {
-		if (!BroadcastSequenceMap.broadcastSequenceMap.containsKey(sceneName)) {
-			return false;
+	public ArrayList<SequenceAction> get(String key, String sceneName) {
+		Map<String, ArrayList<SequenceAction>> map = broadcastSequenceMap.get(sceneName);
+		if (map == null) {
+			map = new HashMap<>();
```

Comment:
Concerning the map changes - I think I got a bit carried away there.
I'll change it back to what you would expect of a map.
Thanks for the elaborate feedback!